apps/src/lib/levelbuilder/code-docs-editor/ProgrammingExpressionEditor.jsx

```diff
@@ -0,0 +1,134 @@
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import TextareaWithMarkdownPreview from '@cdo/apps/lib/levelbuilder/TextareaWithMarkdownPreview';
+import SaveBar from '@cdo/apps/lib/levelbuilder/SaveBar';
+import {navigateToHref} from '@cdo/apps/utils';
+import $ from 'jquery';
+import color from '@cdo/apps/util/color';
+
+function useProgrammingExpression(initialProgrammingExpression) {
+  const [programmingExpression, setProgrammingExpression] = useState(
+    initialProgrammingExpression
+  );
+
+  function updateProgrammingExpression(key, value) {
+    setProgrammingExpression({...programmingExpression, [key]: value});
+  }
+
+  return [programmingExpression, updateProgrammingExpression];
+}
+
+export default function ProgrammingExpressionEditor({
+  initialProgrammingExpression
+}) {
+  // We don't want to update id or key
+  const {
+    id,
+    key,
+    ...remainingProgrammingExpression
+  } = initialProgrammingExpression;
+  const [
+    programmingExpression,
+    updateProgrammingExpression
+  ] = useProgrammingExpression(remainingProgrammingExpression);
+  const [isSaving, setIsSaving] = useState(false);
+  const [lastUpdated, setLastUpdated] = useState(null);
+  const [error, setError] = useState(null);
```

Comment:
I'm not sure if the react 16 stuff has been demoed yet, but if not, this PR could make a nice quick demo at an upcoming eng team meeting. I would say that if you raise enough awareness that other people know to go and look up the react 16 way to do things rather than continuing to do things the old way, that would be a big win!

Reply:
Anjali demoed some stuff last week at demos. Not sure if this adds much to what she demoed though I would definitely be in favor of more exposure to the new features!